getting-started/CONCEPTS/PROMPTS.md

```diff
@@ -0,0 +1,192 @@
+# Prompts
+
+Prompts play a crucial role in communicating and directing the behavior of Large Language Models (LLMs) AI. 
+They serve as inputs or queries that users can provide to elicit specific responses from a model.
+
+Here's a prompt that asks the LLM for name suggestions:
+
+_Input:_
+
+```
+Give me 3 name suggestions for my pet golden retriever.
+```
+
+_Response:_
+
+```
+Some possible name suggestions for a pet golden retriever are:
+
+- Bailey
+- Sunny
+- Cooper
+```
+
+# Prompt Template
+
+Prompt templates are a simple and powerful way to
+define and compose AI functions **using plain text**.
+You can use it to create natural language prompts, generate responses, extract
+information, **invoke other prompts,** or perform any other task that can be
+expressed with text.
+
+The language supports two basic features that allow you to include
+variables and call functions.
+
+**Simple Example:**
+
+Here's an example of a prompt template:
+
+```
+Give me 3 name suggestions for my pet {{ $petName }}.
+```
+
+`$petName` is a variable that is populated on runtime when the template is rendered.
+
+## Prompt Template Language
+
+You don't need to write any code or import any external libraries, just use the
+double curly braces {{...}} to embed expressions in your prompts.
+Teams AI will parse your template and execute the logic behind it.
+This way, you can easily integrate AI into your apps with minimal effort and
+maximum flexibility.
+
+### Variables
+
+To include a variable value in your text, use the `{{$variableName}}` syntax. For example, if you have a variable called name that holds the user's name, you can write:
+
+`Hello {{$name}}, nice to meet you!`
+
+This will produce a greeting with the user's name.
+
+Spaces are ignored, so if you find it more readable, you can also write:
+
+`Hello {{ $name }}, nice to meet you!`
+
+Here's how to define variables in code:
+
+**C#**
+
+In an *action* or *route handler* where the turn state object is available:
+```cs
+state.Temp.Post = "Lorem Ipsium..."
+```
+
+The usage in the prompt:
+```
+This is the user's post: {{ $post }}
+```
+
+> Note: The `turnState.Temp.Post = ...` updates a dictionary with the `post` key under the hood from the [GPT Message Extension sample](https://github.com/microsoft/teams-ai/blob/a20f8715d3fe81e11c330853e3930e22abe298af/dotnet/samples/04.ai.b.messageExtensions.gptME/ActivityHandlers.cs#L156).
```

Comment:
```suggestion
> Note: The `turnState.Temp.Post = ...` updates a dictionary with the `post` key under the hood from the [Message Extension sample](https://github.com/microsoft/teams-ai/blob/a20f8715d3fe81e11c330853e3930e22abe298af/dotnet/samples/04.ai.b.messageExtensions.gptME/ActivityHandlers.cs#L156).
```
We're supposed to avoid using GPT since it's proprietary, I believe. Unless there has been a change?

Reply:
Should also link to the JS sample